mathphys/imgproc.py

```diff
@@ -0,0 +1,1343 @@
+        if saturation_threshold is not None:
+            data[data > saturation_threshold] = saturation_threshold
+        return data, indcs
+
+    @classmethod
+    def generate_gaussian_2d(
+            cls, indcs, sigma=None, mean=None,
+            amplitude=0, offset=0,
+            rand_amplitude=0, saturation_threshold=None,
+            angle=0
+            ):
+        """Generate a bigaussian with given distribution parameters.
+
+        Args:
+            indcs (tuple(2) | list(2) | np.array(2)) :
+                2-component (y and x) pixel index definition. Each component is
+                a (int | tuple | list | np.array) with pixel indice definition.
+            sigma (tuple(2) | list(2) | np.array(2)) :
+                x and y gaussian sigma values (int | float). Defaults to None,
+                corresponding to [numpy.inf]*2
+            mean (tuple(2) | list(2) | np.array(2)) :
+                x and y mean values. Defaults to None, corresponding to
+                [0]*2
+            amplitude (float) : Bigaussian intensity amplitude,
+                Defaults to zero
+            offset (float) : Bigaussian intensity offset. Defaults to zero
+            rand_amplitude (float) : gaussian point intensity random amplitude.
+                Defaults to zero
+            saturation_threshold (float) :
+                Intensity above which image is set to saturated. Defaults to
+                None, in which case no saturation check if performed
+            angle (float) : Bigaussian tilt angles [deg]
+
+        Output:
+            data (np.array) : gaussian curve
+            indcsx (np.array) : x pixel index array (input copy)
+            indcsy (np.array) : y pixel index array (input copy)
+        """
+        # benchmark for size=(1024, 1280)
+        # 35.1 ms ± 833 µs per loop
+        #   (mean ± std. dev. of 7 runs, 10 loops each)
+        indcsx, indcsy = indcs
+        sigmax, sigmay = sigma if sigma else [None] * 2
+        meanx, meany = mean if mean else [None] * 2
+
+        indcsx, sigmax, meanx, \
+            amplitude, offset, rand_amplitude, saturation_threshold = \
+            cls._process_args(
+                indcsx, sigmax, meanx,
+                amplitude, offset, rand_amplitude, saturation_threshold)
+
+        indcsy, sigmay, meany, \
+            amplitude, offset, rand_amplitude, saturation_threshold = \
+            cls._process_args(
+                indcsy, sigmay, meany,
+                amplitude, offset, rand_amplitude, saturation_threshold)
+
+        y = indcsy - meany
+        x = indcsx - meanx
+        mx, my = _np.meshgrid(x, y)
+        angle *= _np.pi / 180  # [deg] -> [rad]
+        cos_a, sin_a = _np.cos(angle), _np.sin(angle)
+        mxl = cos_a * mx - sin_a * my
+        myl = sin_a * mx + cos_a * my
+        data = offset + \
+            amplitude * _np.exp(-0.5 * ((mxl/sigmax)**2 + (myl/sigmay)**2))
```

Comment:
According to this code, sigmax is actually sigma1 and sigmay is sigma2, aren't they?

Reply:
they! I will change the argument naming as to correspond to what they actually are!